### PR TITLE
tidy up and audit gemfile

### PR DESCRIPTION
**Pull request by mksndz**
I suspect that there are some gems included in this project gemfile that aren't currently in use. Removing unused gems can speed up CI build times, reduce update burden, etc. etc.

This PR is intended to start a conversation about which, if any, gems can be removed - as well as tidy things up a bit :)

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,117 +1,65 @@
...
+gem 'solr_wrapper', '0.18.1'
 
+# JRuby related - can we remove?
+gem 'activerecord-jdbcmysql-adapter', '1.3.21', platforms: :jruby
+gem 'jdbc-sqlite3', '3.8.11.2', platforms: :jruby
 
+# why? unicorn gem is also installed...
+# Dockerfile builds from an image with passenger already installed.
 group :production do
   gem 'passenger', '5.1.6', require: 'phusion_passenger/rack_handler'
```
Comment (mksndz) [inline Gemfile:51]: The passenger gem here likely does nothing - we deploy with a Passenger docker image

File: Gemfile [hunk 2]
```diff
@@ -1,117 +1,65 @@
...
-# Use ActiveModel has_secure_password
-# gem 'bcrypt', '~> 3.1.7'
-
-# Use Unicorn as the app server
-gem 'unicorn'
+# potentially removable - or are they used in indexing?
+gem 'jettywrapper', '2.0.4'
+gem 'solr_wrapper', '0.18.1'
 
+# JRuby related - can we remove?
```
Comment (mksndz) [inline Gemfile:44]: @magibney you mentioned the JRuby experiemnt didn't pan out - do you think we can safely remove these gems without impacting Indexing?

File: Gemfile [hunk 3]
```diff
@@ -1,117 +1,65 @@
...
-
-# bundle exec rake doc:rails generates the API under doc/api.
-gem 'sdoc', '0.4.2', group: :doc
-
-# Use ActiveModel has_secure_password
-# gem 'bcrypt', '~> 3.1.7'
-
-# Use Unicorn as the app server
-gem 'unicorn'
+# potentially removable - or are they used in indexing?
```
Comment (mksndz) [inline Gemfile:40]: @magibney @tampakis does anyone make sue of these in development? they should at least be moved in to the `:development` group. Personally I do not use these.

File: Gemfile [hunk 4]
```diff
@@ -1,117 +1,65 @@
...
-
-gem "solr_wrapper", '0.18.1'
-
-gem 'browserify-rails', '4.0.0'
-
-# Use jquery as the JavaScript library
-gem 'jquery-rails', '4.3.1'
-
+# investigate for removal
+gem 'blacklight_range_limit', '6.0.0'
```
Comment (mksndz) [inline Gemfile:35]: @magibney is this a relic from another time? I don't see this plugin used in any Franklin facets

Comment (magibney) [reply Gemfile:35]: Yes, range-limit can be removed. Possibly interested in that at some point, but not used now, nor in the near future.

File: Gemfile (hunk 2 above)
Comment (magibney) [reply Gemfile:44]: jruby stuff can def be removed.

File: Gemfile (hunk 3 above)
Comment (magibney) [reply Gemfile:40]: Not sure what the role of these is; let me try an indexing run with these removed (just to be sure) but I do think they can probably be removed

Comment (magibney) [reply Gemfile:40]: Unless if you know them to be only useful for development, I'm sure I don't use them there...

Comment (mksndz) [reply Gemfile]: i've removed these gem now on this branch - can you test an indexing run and see if there are issues?

**Pull request merged**